File: src/main/java/me/leodev/trollingfreedomreborn/commands/ChatChange.java
package me.leodev.trollingfreedomreborn.commands;

import me.leodev.trollingfreedomreborn.main.Core;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.AsyncPlayerChatEvent;

import java.util.ArrayList;
import java.util.Random;

public class ChatChange implements Listener {

    public static ArrayList<String> Chat1 = new ArrayList();


    public void ChatChange(Player p) {
        String p2 = p.getName();
        Chat1.add(p.getName());
    }

    public void UnChatChange(Player p) {
        String p2 = p.getName();
        Chat1.remove(p.getName());
    }

    @EventHandler
    public void onChat(AsyncPlayerChatEvent event) {
        Player p = event.getPlayer();
        if (Chat1.contains(p.getName())) {
            ArrayList<String> messages = (ArrayList) Core.instance.getConfig().get("troll-config.randomchat");
            Random r = new Random();
            int randomitem = r.nextInt(messages.size());
            String message = messages.get(randomitem);
            String replaced = message.replace("&", "§");
            event.setMessage(replaced);
        }
    }
}

File: src/main/java/com/leomadrassi/trollingfreedomreborn/trolls/chat/ChatChange.kt
package com.leomadrassi.trollingfreedomreborn.trolls.chat

import com.leomadrassi.trollingfreedomreborn.main.Core
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import io.papermc.paper.event.player.AsyncChatEvent
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer

class ChatChange : Listener {
    companion object {
        val Chat1 = mutableListOf<String>()
    }

    fun ChatChange(p: Player) {
        Chat1.add(p.name)
    }

    fun UnChatChange(p: Player) {
        Chat1.remove(p.name)
    }

    @EventHandler
    fun onChat(event: AsyncChatEvent) {
        val p = event.player
        if (Chat1.contains(p.name)) {
            val messages = Core.instance.pluginConfig.getStringList("troll-config.randomchat")
            val randomitem = (0 until messages.size).random()
            val message = messages[randomitem]
            val replaced = message.replace("&", "§")
            val component = LegacyComponentSerializer.legacySection().deserialize(replaced)
            event.message(component)
        }
    }
}
